packages/react-vapor/src/components/textarea/TextArea.tsx

```diff
@@ -57,43 +57,61 @@ const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITextAreaOwnProps): I
     onUnmount: () => dispatch(removeTextArea(ownProps.id)),
 });
 
-export class TextArea extends React.Component<ITextAreaProps, {}> {
-    static defaultProps: Partial<ITextAreaOwnProps> = {
-        additionalAttributes: {},
-        className: '',
+export const TextArea: React.FunctionComponent<ITextAreaProps> = (props) => {
+    const [debouncedValue, setDebouncedValue] = React.useState(props.value);
+    const [isValid, setIsValid] = React.useState(true);
+
+    React.useEffect(() => {
+        setTimeout(() => {
+            setDebouncedValue(props.value);
+        }, 300);
+    }, [props.value]);
+
+    React.useEffect(() => {
+        setIsValid(props.validate?.(debouncedValue));
+    }, [debouncedValue]);
+
+    React.useEffect(() => {
+        props.onMount?.();
+        setIsValid(true);
+        if (props.onUnmount) {
+            return props.onUnmount;
+        }

```

Comment:
`return props.onMount;` no need for the if because the default value is undefined